ring/util.go

```diff
@@ -102,27 +102,28 @@ func WaitRingStability(ctx context.Context, r *Ring, op Operation, minStability,
 
 	// Get the initial ring state.
 	ringLastState, _ := r.GetAllHealthy(op) // nolint:errcheck
-	ringLastStateTs := time.Now()
 
-	const pollingFrequency = time.Second
-	pollingTicker := time.NewTicker(pollingFrequency)
-	defer pollingTicker.Stop()
+	// This timer will be our state
+	stabilityTimer := time.NewTimer(minStability)
+
+	stateWatch, _ := r.statesWatcher.Register()
+	defer r.statesWatcher.UnRegister()
 
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		case <-pollingTicker.C:
-			// We ignore the error because in case of error it will return an empty
-			// replication set which we use to compare with the previous state.
```

Comment:
Why remove this comment, isn't it still relevant?

Reply:
heavy handed, removed the whole case and rebuilt it. fixed